File: C/Colleen.c
#include <stdio.h>

void auto_print(char *str)
{
	printf(str,10,9,34,str);
}

int main(void)
{
	/*
		Inner comment
	*/
	char *str="#include <stdio.h>%1$c%1$cvoid auto_print(char *str)%1$c{%1$c%2$cprintf(str,10,9,34,str);%1$c}%1$c%1$cint main(void)%1$c{%1$c%2$c/*%1$c%2$c%2$cInner comment%1$c%2$c*/%1$c%2$cchar *str=%3$c%4$s%3$c;%1$c%2$cauto_print(str);%1$c}%1$c/*%1$c%2$cOuter comment%1$c*/%1$c";
	auto_print(str);
}
/*
	Outer comment
*/

File: Python/Colleen.py
def auto_print(str):
	print(str % {'nl': 10, 'qu': 34, 'tab': 9, 'string': str})

#Outer comment

def main():
	#Inner comment
	str = "def auto_print(str):%(nl)c%(tab)cprint(str %% {'nl': 10, 'qu': 34, 'tab': 9, 'string': str})%(nl)c%(nl)c#Outer comment%(nl)c%(nl)cdef main():%(nl)c%(tab)c#Inner comment%(nl)c%(tab)cstr = %(qu)c%(string)s%(qu)c%(nl)c%(tab)cauto_print(str)%(nl)c%(nl)cif __name__ == %(qu)c__main__%(qu)c:%(nl)c%(tab)cmain()"
	auto_print(str)

if __name__ == "__main__":
	main()
